### PR TITLE
feat(input):  make "clear input" keyboard interactive

### PR DESCRIPTION
**Pull request by sandgraham**
This PR makes the Clear Icon keyboard navigable and interactive. You can activate the clear by focusing the icon button and pressing Enter or Space. The focus indicator uses our standard visibility checks.

### CODE REVIEW

File: src/input/base-input.js [hunk 1]
```diff
@@ -230,12 +244,21 @@ class BaseInput<T: EventTarget> extends React.Component<
       >
         <ClearIcon
           size={16}
+          tabindex={0}
           title={ariaLabel}
           aria-label={ariaLabel}
           onClick={this.onClearIconClick}
+          onKeyUp={event => {
+            if (event.key && (event.key === 'Enter' || event.key === ' ')) {
+              this.onClearIconClick();
```
Comment (chasestarr) [inline src/input/base-input.js]: you likely need `event.preventDefault()` here to avoid scrolling the page on `space`

Comment (sandgraham) [reply src/input/base-input.js]: also needed to change to keyDown

**Comment by houmark** [comment]
Hey @sandgraham and @chasestarr!

I thought I had commented on this one yesterday, but it seems not.

`tabindex` should be named `tabIndex` as React is throwing a warning for `tabindex` in 9.59.0.

File: src/input/base-input.js [hunk 2]
```diff
@@ -230,12 +244,22 @@ class BaseInput<T: EventTarget> extends React.Component<
       >
         <ClearIcon
           size={16}
+          tabindex={0}
```
Comment (houmark) [inline src/input/base-input.js]: `tabindex` is throwing a warning in React. Should be named `tabIndex`.

**Pull request merged**